1.0/docs/devguide/tests.md

```diff
@@ -0,0 +1,367 @@
+          request = ajax.generateRequest();
+          expect(request.xhr.method).to.be.equal('GET');
+        });
+      });
+    });
+  </script>
+</body>
+</html>
+{% endhighlight %}
+
+## Control which tests are run
+
+Use a test index to run a subset of suites.
+
+{% highlight html %}
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <script src=”../bower_components/webcomponentsjs/web-components-lite.js”></script>
+    <script src=”../bower_components/web-component-tester/browser.js”></script>
+  </head>
+  <body>
+    <script>
+      WCT.loadSuites([
+        'basic.html',
+        'async.html'
+      ]);
+    </script>
+  </body>
+</html>
+{% endhighlight %}
+
+You can also use test indexes to configure your tests via query strings
+when `wct` loads them. See [Test Shadow DOM](#shadow-dom) for an example.
+
+## Prevent shared state with test fixtures {#test-fixtures}
+
+Test fixtures enable you to define a template of content and copy a clean,
+new instance of that content into each test suite. Use test fixtures to
+minimize the amount of shared state between test suites. 
+
+To use a test fixture:
+
+*   Define the test fixture template and give it an ID.
+*   Define a variable in your test script to reference the template.
+*   Instantiate a new instance of the fixture in your `setup()` method.
+
+{% highlight html %}
+<test-fixture id="seed-element-fixture">
+  <template>
+    <seed-element>
+      <h2>seed-element</h2>
+    </seed-element>
+  </template>
+</test-fixture>
+
+<script>
+  suite('<seed-element>', function() {
+    var myEl;
+    setup(function() {
+      myEl = fixture('seed-element-fixture');
+    });
+    test('defines the "author" property', function() {
+      assert.equal(myEl.author.name, 'Dimitri Glazkov');
+    });
+  });
+</script>
+{% endhighlight %}
+
+## Make tests asynchronous
+
+To create an asynchronous test, pass `done` as an argument to the test function
+and then call `done()` when the test is complete. The `done` argument is a
+signal to Mocha that the test is asynchronous. When Mocha runs the tests it 
+waits until it encounters the `done()` call, and eventually times out if it 
+does not encounter it. 
+
+    test('defines the "author" property', function(done) {
+      setTimeout(function() {
+        assert.equal(myEl.author.name, 'Dimitri Glazkov');
+        done();
+      }, 1000);
+    });
+
+## Test Shadow DOM {#shadow-dom}
+
+To test out how a test suite behaves when the browser runs native
+Shadow DOM, create a [test index](#test-indexes) and pass `dom=shadow` as 
+a query string when `wct` loads your test suites.
+
+{% highlight javascript %}
+WCT.loadSuites([
+  'basic-test.html',
+  'basic-test.html?dom=shadow'
+]);
+{% endhighlight %}
+
+Within your tests, you can access Shadow DOM elements using Polymer's DOM 
+API.
+
+    test('click sets isWaiting to true', function() {
+      Polymer.dom(myEl.root).querySelector('button').click();
```

Comment:
Alternatively, it looks like you can also do this:

`myEl.$$('button')` which is much easier. Wish I'd know that when I did the video :P
https://www.polymer-project.org/1.0/docs/devguide/utility-functions.html